### PR TITLE
refactor: further shell script linting

### PR DESCRIPTION
**Pull request by MitchellPaff**
Done the obvious. Any suggestions on further changes appreciated.

### CODE REVIEW

File: scripts/linz-bde-schema-load.bash [hunk 1]
```diff
@@ -84,21 +102,24 @@ dbpatch-loader - fake 2>&1 | grep -q "database.*does not exist" &&
     DBPATCH_SUPPORTS_STDOUT=no ||
     DBPATCH_SUPPORTS_STDOUT=yes
 
-if test $PGDATABASE = "-" -a $DBPATCH_SUPPORTS_STDOUT != yes; then
+if test "$PGDATABASE" = "-" -a $DBPATCH_SUPPORTS_STDOUT != yes
```
Comment (l0b0) [inline scripts/linz-bde-schema-load.bash:105]: `$DBPATCH_SUPPORTS_STDOUT` should also be quoted. If it's not, and the value is ever empty, then the expression gets borked. Ditto for other unquoted variable references.

File: scripts/linz-bde-schema-load.bash [hunk 2]
```diff
@@ -138,38 +162,44 @@ fi
...
 cat << EOF
 CREATE EXTENSION IF NOT EXISTS postgis SCHEMA public;
 EOF
 
-for file in ${SCRIPTSDIR}/*.sql; do
+for file in "${SCRIPTSDIR}"/*.sql
+do
     if test ${SKIP_INDEXES} = 'yes' &&
-       `basename $file .sql` = '04-bde_schema_index';
+       [ "$(basename "$file" .sql)" = '04-bde_schema_index' ]
```
Comment (l0b0) [inline scripts/linz-bde-schema-load.bash:188]: Ouch, this looks like an actual bug in the original code. It should probably be a separate PR.

File: test/test-publication.bash [hunk 3]
```diff
@@ -0,0 +1,39 @@
+#!/usr/bin/env bash
```
Comment (l0b0) [inline test/test-publication.bash:1]: It might be good to split the rename (and shebang line) into a separate PR, to make it easier to see changes over time.

File: scripts/linz-bde-schema-load.bash (hunk 1 above)
Comment (MitchellPaff) [reply scripts/linz-bde-schema-load.bash:105]: Does this apply only when the '$' appears at the beginning of the variable or every occurrence?

Comment (MitchellPaff) [reply scripts/linz-bde-schema-load.bash:105]: e.g should DBPATCH_SUPPORTS_STDOUT on lines 102/103 be quoted?

Comment (l0b0) [reply scripts/linz-bde-schema-load.bash:105]: Lines 102/103 are *assignments.* You only need to quote variable *references.*

File: scripts/linz-bde-schema-load.bash (hunk 2 above)
Comment (MitchellPaff) [reply scripts/linz-bde-schema-load.bash:188]: How so? This change is required for Shellcheck to pass the https://www.shellcheck.net/wiki/SC2283  check

Comment (l0b0) [reply scripts/linz-bde-schema-load.bash:186]: The original code was

```shell
     if test ${SKIP_INDEXES} = 'yes' &&
       `basename $file .sql` = '04-bde_schema_index'; 
```

To see the problem, we can simplify that to `test yes = 'yes' && 04-bde_schema_index = '04-bde_schema_index'`. But that fails, because the `&&` is a command separator, and doesn't work the same as `-a` for the `test` command:

```console
$ test yes = 'yes' && 04-bde_schema_index = '04-bde_schema_index'
04-bde_schema_index: command not found
```

So the original code was broken, in the sense that the second line would never be true.

**Review by l0b0 — APPROVED** [review]
